rfcs/require-by-string.md

```diff
@@ -0,0 +1,360 @@
+
+Instead, absolute paths should appear only in the alias map. Then, only the alias map will need to be modified if a package's location is changed on disk, and in the future, package management software could automatically manage the alias map rather than needing to reach into and modify individual Luau files.
+
+#### Aliases
+
+Aliases can be used to bind an absolute or relative path to a convenient name that can be required directly. In Luau files, aliases will always be prefixed with `@` to unambiguously distinguish them from directory names.
+
+```json
+"aliases": {
+    "Roact": "C:/LuauModules/Roact-v1.4.2"
+}
+```
+
+Based on the alias map above, you would be able to require Roact directly:
+
+```lua
+local Roact = require("@Roact")
+```
+
+Or even a sub-module:
+
+```lua
+local createElement = require("@Roact/createElement")
+```
+
+Note that the alias map itself does not contain any `@` prefixes, which differs from some other languages. For example, when configuring paths in `tsconfig.json` for TypeScript projects, the `@` prefix is optional. Instead, the alias name is matched exactly in require statements. To reduce ambiguity in Luau files, we will always require `@` to be prepended to the alias name that appears in the alias map.
+
+##### Versioning
+
+Aliases are simple bindings and aren't concerned with versioning. The intention is to allow package management software to leverage aliases by automatically adding and updating the alias map to reflect a package's dependencies.
+
+##### Library root alias
+
+In the past, it has been proposed to use the blank alias "`@`" to represent the root directory of a file's encapsulating library. 
+- However, the concept of a "Luau library" and its root directory is not yet rigorously defined in Luau, in terms of folder/file structure. 
+- In the future, we may add a `package.json` file or something similar that marks the root directory of a library, but this is outside of the scope of this RFC, which primarily focuses on improving require-by-string.
+- For the time being, this functionality will remain unimplemented for this reason. The blank alias "`@`" will remain reserved for now, meaning it cannot be overriden.
+
+Of course, users can still use the alias map to explicitly define this behavior with a named alias:
+
+```json
+"aliases": {
+    "src": "../"
+}
+```
+
+##### Limitations of aliases
+
+- Aliases cannot reference other aliases.
+- Alias names cannot contain the directory separators `/` and `\`.
+- All aliases must be prefixed with `@` when used in a require statement to avoid ambiguity.
+- Aliases can only occur at the beginning of a path.
+
+##### Configuring alias maps: .luaurc
+
+As part of this proposal, alias maps will be configured in `.luaurc`, which follows a JSON-like syntax.
```

Comment:
Can alias refer to an alias? E.g.:

```
@Roact => @game/Packages/Roact-v12
@game => DM root (or absolute FS path)
```

Reply:
This was left out on purpose in a previous iteration of this RFC, as it is easy to add in later but would be difficult to remove after implementing. If we don't see any reason not to add this though, then I'm happy adding this to the RFC as well.

> ```
> @Roact => @game/Packages/Roact-v12
> @game => DM root (or absolute FS path)
> ```


I assume we would want aliases to be defined in order? E.g. the example you give above would fail, as `@Roact` refers to `@game`, which isn't defined yet (as it appears second).

Reply:
Right in my example I assumed these get defined in different .luaurc files. I think this is fine to omit out of the initial RFC but we should add a note, e.g say that the path alias resolves to currently can not contain aliases but this will be added in the future, or smth to that effect.